packages/flutter/lib/src/widgets/sensitive_content.dart

```diff
@@ -0,0 +1,277 @@
+    _contentSenstivityIsSupported ??= await _sensitiveContentService.isSupported();
+    if (!_contentSenstivityIsSupported!) {
+      // Setting content sensitivity is not supported on this device.
+      return;
+    }
+
+    // If needed, set default content sensitivity level as set in native Android. This will be
+    // auto sensitive if it is otherwise unset by the developer. Also, initialize the current
+    // content sensitivity level if needed.
+    _defaultContentSensitivitySetting ??= await _sensitiveContentService.getContentSensitivity();
+    currentContentSensitivityLevel ??= _defaultContentSensitivitySetting;
+
+    // Update SensitiveContent widget count for those with desiredSensitivityLevel.
+    _contentSensitivitySetting.addWidgetWithContentSensitivity(desiredSensitivityLevel);
+
+    // Verify that desiredSensitivityLevel should be set in order for sensitive
+    // content to remain obscured.
+    if (!shouldSetContentSensitivity(desiredSensitivityLevel)) {
+      return;
+    }
+
+    // Set content sensitivity level as desiredSensitivityLevel. If the call to set content
+    // sensitivity on the platform side fails, then we do not update the current content
+    // sensitivity level.
+    _sensitiveContentService.setContentSensitivity(desiredSensitivityLevel);
+
+    // Update current content sensitivity level.
+    currentContentSensitivityLevel = desiredSensitivityLevel;
+  }
+
+  /// Unregisters a [SensitiveContent] widget from the [ContentSensitivitySetting] tracking
+  /// the content sensitivity level of the widget tree.
+  static void unregister(ContentSensitivity widgetSensitivityLevel) {
+    instance._unregister(widgetSensitivityLevel);
+  }
+
+  void _unregister(ContentSensitivity widgetSensitivityLevel) {
+    assert(
+      _contentSenstivityIsSupported != null,
+      'SensitiveContentHost.register must be called before SensitiveContentHost.unregister',
+    );
+    if (!_contentSenstivityIsSupported!) {
+      // Setting content sensitivity is not supported on this device.
+      return;
+    }
+
+    // Update SensitiveContent widget count for those with
+    // desiredSensitivityLevel.
+    _contentSensitivitySetting.removeWidgetWithContentSensitivity(widgetSensitivityLevel);
+
+    if (!_contentSensitivitySetting.hasWidgets) {
+      // Restore default content sensitivity setting if there are no more SensitiveContent
+      // widgets in the tree. If the call to set content sensitivity on the platform side fails,
+      // then we do not update the current content sensitivity level.
+      _sensitiveContentService.setContentSensitivity(_defaultContentSensitivitySetting!);
+
+      // Update current content sensitivity level.
+      currentContentSensitivityLevel = _defaultContentSensitivitySetting;
+      return;
+    }
+
+    // Determine if another sensitivity level needs to be restored.
+    late final ContentSensitivity? contentSensitivityToRestore =
```

Comment:
I dont think this needs a late?  (or if it does then maybe it doenst need to be final or nullable but all three feels weird) Especially because it is being given a value on the right hand side of =.

Reply:
the late here will make it a lazy initializer. so if the platform doesn't support is contentsensitivity, we won't create this at all (thus save some memory)

Reply:
I think the `late final` method was for `_contentSensitivitySetting`. I think this was an accidental edit that was caught while I was still messing around with this file; removed.